src/login/ResetPasswordPage.tsx

```diff
@@ -136,23 +137,34 @@ const ResetPasswordPage = () => {
     }
 
     resetPassword(resetPwdData).then((response) => {
-      if ("error" in response) {
-        const receivedError = response.error as MetaResponse;
-        const reason = receivedError.response?.headers.get(
+      const { response: resp, metaResponse } =
+        response.error as ResponseOnPwdReset;
+
+      if (resp !== undefined && metaResponse !== undefined) {
+        const reason = metaResponse?.response?.headers.get(
           "x-ipa-pwchange-result"
         );
+        const htmlMessage = resp.data as string;
+
+        const match = htmlMessage.match(/<strong>(.*?)<\/strong>/);
 
-        if (reason === "invalid-password") {
-          alerts.addAlert(
-            "reset-password-error",
-            "The password or username you entered is incorrect",
-            "danger"
-          );
-          clearFields();
-          setBtnSpinning(false);
-        } else {
-          // Redirect to login page to allow the user to login with new credentials
-          navigate("/login");
+        if (match && match[1]) {
+          const errorMessage = match[1];
+          if (errorMessage.includes("Password is too short")) {
+            alerts.addAlert("reset-password-error", errorMessage, "danger");
+            setBtnSpinning(false);
+          } else if (reason === "invalid-password") {
+            alerts.addAlert(
+              "reset-password-error",
+              "The password or username you entered is incorrect",
+              "danger"
+            );
+            clearFields();
+            setBtnSpinning(false);
+          } else {
+            // Redirect to login page to allow the user to login with new credentials
+            navigate("/login");
```

Comment:
Shouldn't we instead show a generic error containing the message from the server? Something like 

```suggestion
            alerts.addAlert(
              "reset-passwords-error",
              reason,
              "danger"
            );
```

Reply:
Oh, I see, but the navigate is run only when `reason` is empty right? Therefore we can just check whether it's empty, if so, redirect, otherwise match known errors or display an unknown one?

Reply:
> Oh, I see, but the navigate is run only when `reason` is empty right? 
I think so? Might need to check, this API is a bit odd on its responses...

> Therefore we can just check whether it's empty, if so, redirect, otherwise match known errors or display an unknown one?
That make sense. Thing is, there are no many errors that can be found in the reset pages (apart from 'password is too short'  or ' passwords does not match'), but I can add what you are suggesting for any other errors just in case.

Reply:
> It's up to you, I'd rather do it for other errors just to save my future self the pain, but it's fine to keep it as it's then.

I added an extra check to the code to manage any other error coming from the call.